Generated library's ng-package.json file creates the wrong output file path reference for the build system

## I'm submitting a...
<pre><code>
[ x ] Bug report 
</code></pre>

## Current behavior
After building a library using the Angular CLI, app cannot find a reference to library when importing in app.module.ts

## Expected behavior
It finds a proper reference to MyTestLibrary.

## Minimal reproduction of the problem with instructions
Create a library containing capital letters like MyTestLibrary with the command
<code>ng g library MyTestLibrary</code>

Build MyTestLibrary with command <code>ng build MyTestLibrary</code>
Import library in app.module.ts with <code>import { MyTestLibraryComponent } from 'MyTestLibrary';</code>
*see error: can not find module MyTestLibrary*
## What is the motivation / use case for changing the behavior?
The Angular CLI is not providing a proper reference to where the build system outputs the library.

I found the library's generated ng-package.json file is using the libary's hyphenated form instead of the actual name of the library. The generator is also providing a file reference so you can actually reference your built library in tsconfig.json. These 2 file paths are differing. That's the problem. The build system will output your library to dist/my-test-library while your app only has an expected reference of dist/MyTestLibrary.

## Environment
Any

<pre><code>
Angular CLI version: 6.0.0

@filipesilva All libraries generated with 6.1.3. cannot be imported into the app using just the name. Example: `ng g library my-lib`. It is not possible to `import { MyModule } from 'my-lib'`, it will raise an error `Cannot find module 'my-lib'`

I'm not quite sure how your link references this issue. The partial build feature is def appreciated though. If you are getting `Cannot find module 'my-lib'`, and you are SURE it should be working, I typically ignore it or close and reopen vs code. You can also run `ng build` to make sure.

--- a/packages/schematics/angular/library/index.ts
+++ b/packages/schematics/angular/library/index.ts
@@ -131,7 +131,7 @@ function addDependenciesToPackageJson() {
 }
 
 function addAppToWorkspaceFile(options: LibraryOptions, workspace: WorkspaceSchema,
-                               projectRoot: string, packageName: string): Rule {
+                               projectRoot: string, projectName: string): Rule {
 
   const project: WorkspaceProject<ProjectType.Library> = {
     root: projectRoot,
@@ -169,7 +169,7 @@ function addAppToWorkspaceFile(options: LibraryOptions, workspace: WorkspaceSche
     },
   };
 
-  return addProjectToWorkspace(workspace, packageName, project);
+  return addProjectToWorkspace(workspace, projectName, project);
 }
 
 export default function (options: LibraryOptions): Rule {
@@ -182,7 +182,8 @@ export default function (options: LibraryOptions): Rule {
     validateProjectName(options.name);
 
     // If scoped project (i.e. "@foo/bar"), convert projectDir to "foo/bar".
-    const packageName = options.name;
+    const projectName = options.name;
+    const packageName = strings.dasherize(projectName);
     let scopeName = null;
     if (/^@.*\/.*/.test(options.name)) {
       const [scope, name] = options.name.split('/');
@@ -218,7 +219,7 @@ export default function (options: LibraryOptions): Rule {
 
     return chain([
       branchAndMerge(mergeWith(templateSource)),
-      addAppToWorkspaceFile(options, workspace, projectRoot, packageName),
+      addAppToWorkspaceFile(options, workspace, projectRoot, projectName),
       options.skipPackageJson ? noop() : addDependenciesToPackageJson(),
       options.skipTsConfig ? noop() : updateTsConfig(packageName, distRoot),
       schematic('module', {

